src/threads.cc

```diff
@@ -0,0 +1,153 @@
+#include <dlfcn.h>
+#include <cstddef>
+#include <cstdlib>
+
+#include <iostream>
+
+#include <boost/algorithm/string/join.hpp> // for boost::algorithm::join
```

Comment:
Do we need to update http://developer.lsst.io/en/latest/coding/using_boost.html ?